unidist/core/backends/dask/backend.py

```diff
@@ -151,6 +153,26 @@ def num_cpus():
         """
         return len(get_client().ncores())
 
+    @staticmethod
+    def cluster_resources():
+        """
+        Get resources of Dask cluster.
+
+        Returns
+        -------
+        dict
+            Dictionary with cluster nodes info in the form '{node_ip0: {CPU: x0},
+            node_ip1: {CPU: x1}, ..}'.
+        """
+        client = get_client()
+        cluster_resources = defaultdict(lambda: {"CPU": 0})
+        for worker_info in client.scheduler_info()["workers"].values():
+            cluster_resources[worker_info["host"]]["CPU"] += 1
```

Comment:
We treat `num_cpus` as a parameter indicating the number of CPUs used by the execution backend to schedule tasks. In depend on a concrete execution backend `num_cpus` may and may not match the number of workers used, for example, for Dask.

Reply:
In that case we are breaking unified execution.

For example, user has a cluster with 2 nodes, each node has 4 cores and user decides to use only by 2 cores per node on cluster needs. In case of Ray backend the output will be:
```python
>>ud.cluster_resources()
{localhost: {CPU: 2}, other_host: {CPU: 2}}
```
And user can implement logic assuming on values of "CPU" resources and believing in execution of 4 workers at the same time.

In case of using Dask backend with the way you proposed, "CPU" resources strongly depends from user's settings during Dask cluster enabling:
User created 2 workers on each node without specifying the number of threads per worker. In this case all threads of node will be divided between workers on node and behavior will be next:
```python
>>ud.cluster_resources()
{localhost: {CPU: 4}, other_host: {CPU: 4}} // Because each worker has 2 threads
```

In this case the logic, implemented by user for Ray backend won't work for Dask, because "CPU" here is a number of threads, not workers. It can confuse the user with number of tasks executed at the same time.

We must provide backend-agnostic API with expected similar behavior for every backend.

Reply:
Discussed offline: it was decided to provide real number of threads from dask process. Implemented in accordance with the discussion.